custom/src/main/java/com/solarwinds/opentelemetry/extensions/initialize/config/HttpSettingsReaderDelegate.java

```diff
@@ -0,0 +1,98 @@
+
+  private static final Logger logger = LoggerFactory.getLogger();
+
+  private static final Gson gson = new GsonBuilder().create();
+
+  public Settings fetchSettings(String url, String authorizationHeader) {
+    AtomicReference<Settings> settings = new AtomicReference<>();
+    InstrumentationUtil.suppressInstrumentation(
+        () -> {
+          HttpURLConnection connection = null;
+          try {
+            connection = getHttpURLConnection(new URL(url), authorizationHeader);
+            int responseCode = connection.getResponseCode();
+            if (responseCode != HttpURLConnection.HTTP_OK) {
+              String errorResponse = getErrorMessage(connection);
+              logger.error(
+                  String.format(
+                      "HTTP request failed with status code: %s and error: %s",
+                      responseCode, errorResponse));
+            } else {
+              String payload;
+              try (BufferedReader reader =
+                  new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
+                StringBuilder responseBuilder = new StringBuilder();
+
+                String line;
+                while ((line = reader.readLine()) != null) {
+                  responseBuilder.append(line);
+                }
+                payload = responseBuilder.toString();
+              }
+
+              settings.set(JsonSettingWrapper.wrap(gson.fromJson(payload, JsonSetting.class)));
+            }
+
+          } catch (IOException e) {
+            logger.error(String.format("IO error while fetching settings from URL: %s", url), e);
+
+          } catch (Exception e) {
+            logger.error(String.format("Failed to get settings from http URL: %s", url), e);
+          } finally {
+            if (connection != null) {
+              connection.disconnect();
+            }
+          }
+        });
+
+    return settings.get();
+  }
+
+  private String getErrorMessage(HttpURLConnection connection) {
+    String errorResponse;
+    try (BufferedReader errorReader =
+        new BufferedReader(new InputStreamReader(connection.getErrorStream()))) {
+      StringBuilder errorBuilder = new StringBuilder();
+      String line;
+      while ((line = errorReader.readLine()) != null) {
+        errorBuilder.append(line);
+      }
+      errorResponse = errorBuilder.toString();
+    } catch (Exception e) {
+      errorResponse = "Unable to read error response";
+    }
+    return errorResponse;
+  }
+
+  HttpURLConnection getHttpURLConnection(URL url, String authorizationHeader) throws IOException {
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+
+    connection.setRequestMethod("GET");
+    connection.setRequestProperty("Authorization", authorizationHeader);
+    connection.setRequestProperty("Content-Type", "application/json");
+
+    connection.setRequestProperty("Accept", "application/json");
+    connection.setConnectTimeout(60000);
```

Comment:
Curious any reason for a 60s connection timeout? That seems pretty long and since same as polling interval, wondering if any negative consequences.

Reply:
no reason just enough time to establish connection. i can dial it down.